Orange/widgets/utils/itemmodels.py

```diff
@@ -849,10 +865,17 @@ def columnSortKeyData(self, column, role):
         coldesc = self.columns[column]
         if isinstance(coldesc, TableModel.Column) \
                 and role == TableModel.ValueRole:
-            col_view, _ = self.source.get_column_view(coldesc.var)
+            var = coldesc.var
+            col_view, _ = self.source.get_column_view(var)
             col_data = numpy.asarray(col_view)
             if self.__sortInd is not None:
                 col_data = col_data[self.__sortInd]
+            value_keys = self.__sortValueKeys[column]
+            if value_keys is not None:
+                if col_data is col_view:
+                    col_data = numpy.array(col_data)
+                col_data[numpy.isnan(col_data)] = len(var.values)
+                col_data = value_keys[col_data.astype(numpy.int)]
```

Comment:
Sorting discrete floats could work if above cast was made to `float` and here `np.searchsorted(value_keys, col_data)`?